File: src/main/java/com/zerozero/image/domain/model/ImagePrefix.java
package com.zerozero.image.domain.model;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Arrays;

@Getter
@RequiredArgsConstructor
public enum ImagePrefix {
    STORE("store"),
    USER("user"),
    ;

    private final String prefix;

    public static ImagePrefix fromString(String value) {
        return Arrays.stream(ImagePrefix.values())
                .filter(imagePrefix -> imagePrefix.getPrefix().equalsIgnoreCase(value))
                .findFirst()
                .orElse(null);
    }
}

File: src/main/java/com/zerozero/image/domain/model/ImagePrefix.kt
package com.zerozero.image.domain.model

enum class ImagePrefix {
    STORE,
    USER,
    ;

    companion object {
        fun fromString(value: String): ImagePrefix? {
            return entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
        }
    }
}
